CMakeLists.txt

```diff
@@ -73,6 +74,20 @@ foreach (protobuf_file ${MESSAGE_PROTOS_COMMON})
         PROPERTIES GENERATED TRUE)
 endforeach()
 
+##########################################################
+# component proto files
+##########################################################
+# generate source from component proto files from toplevel to get correct header file guard 
+
+# collect proto files from sub-directories below directory component
+FILE(GLOB_RECURSE MESSAGE_PROTOS_COMPONENTS ABSOLUTE ${CMAKE_CURRENT_SOURCE_DIR}/components *.proto)
+# create single library with all generated sources from component proto files
+add_library(MRA-components-proto)
```

Comment:
Hm, for the long run, this will become a build hotspot. But ok for now I guess.

Reply:
Indeed, it is not preferred. But will solve the issue of creating correct header files.
It is only used for the generating of code from proto.